File: C_Codes/Ponteiros/stringManipulation.c
#include <stdio.h>

void to_uppercase(char *str) {
    while (*str) {
        if (*str >= 'a' && *str <= 'z') {
            *str = *str - ('a' - 'A');
        }
        str++;
    }
}

int main() {
    char str[] = "hello, world!";
    to_uppercase(str);
    printf("Uppercase: %s\n", str);
    return 0;
}

File: Códigos/Python Codes/Ponteiros/string_manipulation.py
def to_uppercase(s):
    """Converts a string to uppercase manually."""
    result = []
    for char in s:
        if 'a' <= char <= 'z':  # Verifica se o caractere está entre 'a' e 'z'
            # Converte o caractere para maiúscula
            result.append(chr(ord(char) - (ord('a') - ord('A'))))
        else:
            result.append(char)
    return ''.join(result)


def main():
    text = "hello, world!"
    uppercase_text = to_uppercase(text)
    print(f"Uppercase: {uppercase_text}")


if __name__ == "__main__":
    main()
